client/client.h

```diff
@@ -48,22 +48,85 @@ class Client {
+  ///     (default = -1), the value is interpreted as follows:
+  ///    -1 = blocking send operation
+  ///     0 = non-blocking send operation without retries
+  ///    >0 = time (in milliseconds) after which the function returns an error,
+  ///         if the send operation was not accomplished
+  /// @param receive_timeout_ms [in] Receive operation timeout in milliseconds
+  ///     (default = -1), the value is interpreted as follows:
+  ///    -1 = blocking receive operation
+  ///     0 = non-blocking receive operation without retries
+  ///    >0 = time (in milliseconds) after which the function returns an error,
+  ///         if the receive operation was not accomplished
+  /// @return true if the connection was established; false otherwise
+  bool connect(const std::string& hostname, int port,
+      int send_timeout_ms = -1, int receive_timeout_ms = -1);
+
+  /// Indicates whether the connection was successfully established
+  /// @return true if the connection was successfully established;
+  ///     false otherwise
   bool connected() const {
     return conn_ != nullptr;
   }
+
+  /// Close the socket connection if it was previously successfully established
+  /// and destroy the associated socket
+  /// @return true if the connection was closed successfully; false if the
+  ///     connection did not exist
   bool close();
+
+  /// Perform handshake over the established connection.
+  /// @param updates [out] State field names that were updated from
+  ///     the handshake response
+  /// @param opts [in] Options to pass in the handshake message
+  ///     (default = Options())
+  /// @return true if the handshake succeeded; false otherwise
   bool init(std::vector<std::string>& updates, const Options& opts = Options());
+
+  /// Send a message containing commands over the established socket connection
+  /// @param commands [in] Commands to send over the socket connection
+  /// @return false if one of the following conditions holds:
+  ///     1) this is a second send operation in a row (without an
+  ///        intermediate receive operation);
+  ///     2) the socket connection has not been successfully established
+  ///     3) send operation failed due to an error or a timeout
+  ///     Otherwise, returns true
   bool send(const std::vector<Command>& commands);
+
+  /// Receive a message containing state updates over the established socket
+  /// connection
+  /// @param updates [out] State field names that were updated from
+  ///     the received message
+  /// @return false if one of the following conditions holds:
+  ///     1) this is a second receive operation in a row (without an
+  ///        intermediate send operation);
+  ///     2) the socket connection has not been successfully established
+  ///     3) receive operation failed due to an error or a timeout
+  ///     4) unknown message type was encountered
+  ///     Otherwise, returns true
```

Comment:
The individual error reasons for these functions are all accessible via `error()` after the fact. I guess it would make sense to refer to that function in the documentation so that it doesn't get out of sync with the code so easily.